src/include/daos_fs.h

```diff
@@ -253,6 +253,36 @@ int
 dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor,
 	    uint32_t *nr, struct dirent *dirs);
 
+/**
+ * User callback defined for dfs_readdir_size.
+ */
+typedef int (*dfs_readdir_cb_t)(dfs_t *dfs, dfs_obj_t *obj, const char name[],
+				void *_udata);
+
+/**
+ * Same as dfs_readdir, but this also adds a buffer size limitation when
+ * enumerating. On every entry, it issues a user defined callback. If size
+ * limitation is reached, function returns -DER_KEY2BIG.
```

Comment:
should probably use errno errors instead now.

Reply:
yes i will address this when that patch lands and i merge, or vice versa.